layout/custompaddedlayout.go

```diff
@@ -0,0 +1,74 @@
+package layout
+
+import (
+	"fyne.io/fyne/v2"
+)
+
+// Declare conformity with Layout interface
+var _ fyne.Layout = (*CustomPaddedLayout)(nil)
+
+// CustomPaddedLayout is a layout similar to PaddedLayout, but uses
+// custom values for padding on each side, rather than the theme padding value.
+//
+// Since: 2.5
+type CustomPaddedLayout struct {
+	TopPadding    float32
+	BottomPadding float32
+	LeftPadding   float32
+	RightPadding  float32
+}
+
+// NewCustomPaddedLayout creates a new CustomPaddedLayout instance
+// with the specified paddings.
+//
+// Since: 2.5
+func NewCustomPaddedLayout(padTop, padBottom, padLeft, padRight float32) fyne.Layout {
+	return CustomPaddedLayout{
+		TopPadding:    padTop,
+		BottomPadding: padBottom,
+		LeftPadding:   padLeft,
+		RightPadding:  padRight,
+	}
+}
+
+// NewSquareCustomPaddedLayout creates a new CustomPaddedLayout instance
+// with the same custom padding applied to all four sides.
+//
+// Since: 2.5
+func NewSquareCustomPaddedLayout(pad float32) fyne.Layout {
```

Comment:
Hmm, not sure about the naming here as we have used "SquareSize" to indicate equal width and height.
Clearly this is not creating square content. Is there a better name?

Reply:
Nothing great is coming to mind (NewEqualCustomPaddedLayout maybe, but it's a bit awkward). So I'll remove this shorthand for now so we can land it